com.microsoft.mrtk.input/Visualizers/ControllerVisualizer/ControllerVisualizer.cs

```diff
@@ -0,0 +1,191 @@
+                    ClearControllerVisuals();
+                }
+            }
+        }
+
+        private GameObject ControllerGameObject;
+        /// <summary>
+        /// Tries to instantiate controller visuals for the specified hand node.
+        /// </summary>
+        /// <param name="useFallbackVisuals">Whether or not to use the fallback controller visuals</param>
+        private async void InstantiateControllerVisuals(bool usePlatformVisuals, bool useFallbackVisuals)
+        {
+            // Initialize the ControllerGameObject if it has not yet been initialized
+            // First try to load the controller model from the platform
+            if (ControllerGameObject == null)
+            {
+                if (usePlatformVisuals && controllerModelProvider != null)
+                {
+                    GameObject PlatformLoadedGameObject = await ControllerModelLoader.TryGenerateControllerModelFromPlatformSDK(controllerModelProvider);
+                    if (PlatformLoadedGameObject != null)
+                    {
+                        // Platform models are "rotated" 180 degrees due to the forward vector for a controller pointing towards the user.
+                        // We need to rotate these models in order to have them pointing in the correct direction on device
+                        GameObject rotationAdjustedGameObject = new GameObject(PlatformLoadedGameObject.name + " Root");
+                        PlatformLoadedGameObject.transform.parent = rotationAdjustedGameObject.transform;
+                        PlatformLoadedGameObject.transform.SetPositionAndRotation(rotationAdjustedGameObject.transform.position, rotationAdjustedGameObject.transform.rotation * Quaternion.Euler(0, 180, 0));
+                        ControllerGameObject = rotationAdjustedGameObject;
+                    }
+                }
+
+                // If the ControllerGameObject is still not initialized after this, then use the fallback model if told to
+                if (ControllerGameObject == null && useFallbackVisuals)
+                {
+                    ControllerGameObject = Instantiate(fallbackControllerModel);
+                }
+            }
+
+            if (ControllerGameObject != null)
+            {
+                ControllerGameObject.transform.parent = transform;
+                ControllerGameObject.transform.SetPositionAndRotation(transform.position, transform.rotation);
+            }
+        }
+
+        /// <summary>
+        /// Clears the controller visuals if they have been instantiated
+        /// </summary>
+        private void ClearControllerVisuals()
+        {
+            if (ControllerGameObject != null)
+            {
+                Destroy(ControllerGameObject);
+            }
+        }
+
+        public void Update()
+        {
+            if(ControllerGameObject != null && ControllerGameObject.activeSelf != IsControllerTracked)
```

Comment:
nit format

Reply:
Should be fixed if not outdated by a bit!